coralproject/talk
# [CORL-2413]: Add site search to site moderation modal

## What does this PR do?

This PR updates the site moderator modal for admins to use the site search. In addition, this fixes a bug where if the user selected no sites to assign, the org moderator role would be assigned; instead, the `Assign` button is now disabled if no sites are selected in this modal. These changes are for https://vmproduct.atlassian.net/browse/CORL-2413. 

## What changes to the GraphQL/Database Schema does this PR introduce?

None

## How do I test this PR?

Log in as an admin. Click on the `Role` change dropdown. Select `Site moderator`. See that if a user already has sites that they can moderate, these are displayed as selected checkboxes. See the site search. If you search and add sites, they should be added to the selected checkboxes. You should be able to select/deselect these checkboxes once you've added a site. Click assign. See that the selected sites are assigned for site moderation for that user.

Also check that if you have no sites selected in the site moderator modal, the `Assign` button is disabled (greyed out) until you have selected at least one site to assign permissions for.

See that if you are logged in as a site moderator and you select the `Role` change dropdown, you still see the site moderator's list of scoped sites to apply to the user. (There's a separate issue open to address making these checkboxes instead.)
 
## How do we deploy this PR?
